### PR TITLE
[WIP] Don't override boot order on Template clone

### PR DESCRIPTION
**Pull request by matobet**
Bug-Url: https://bugzilla.redhat.com/show_bug.cgi?id=1340163

### CODE REVIEW

**Comment by oourfali** [comment]
Change looks good to me.
@gmcculloug  can you review as well?
Not sure why there are security warnings and build failures, doesn't seem related but I'm not familiar enough with it.

**Comment by matobet** [comment]
@miq-bot add-label darga/yes

**Comment by gmcculloug** [comment]
@syncrou @bdunne Please review.

**Comment by syncrou** [comment]
There seem to be some test failures due to the change.  Aside from that - I ran a test with this change using my multiple boot device template - and it provisioned correctly.

So - (minus the failing tests) :+1: LGTM

File: lib/ovirt/template.rb [hunk 1]
```diff
@@ -152,9 +152,7 @@ def build_clone_xml(options)
             end
           end
 
-          xml.os(:type => options[:os_type] || os_type) do
-            xml.boot(:dev => 'hd')
-          end
+          xml.os(:type => options[:os_type] || os_type)
```
Comment (gmcculloug) [inline lib/ovirt/template.rb]: @bdunne Will this work when cloning uses the blank_template?  [lib/ovirt/template.rb#L109](https://github.com/ManageIQ/ovirt/blob/7184f6f36401cb00d4b233c60dbbc3f90cf3feb1/lib/ovirt/template.rb#L109)

**Comment by matobet** [comment]
Fixed failing tests.

File: lib/ovirt/template.rb (hunk 1 above)
Comment (bdunne) [inline lib/ovirt/template.rb]: I don't think so, I would expect it to take the defaults from the blank template.  @metobet Can you test this?

Comment (matobet) [inline lib/ovirt/template.rb]: @gmcculloug @bdunne Yes, indeed. When the boot device(s) are missing, the default is correctly taken from the template - even the blank one; in which case it is `hd`. Also tested it just to be sure :+1:

Comment (bdunne) [inline lib/ovirt/template.rb]: @matobet When provisioning PXE or ISO, I believe we always clone from the blank template then modify based on the selected source.  I think we still have the same problems in these cases.

Comment (matobet) [inline lib/ovirt/template.rb]: @bdunne not sure if I follow. When creating vm from template the boot order is simply copied. Of course, you can always edit it afterwards.

Comment (bdunne) [inline lib/ovirt/template.rb]: When running a PXE or ISO provision we use the skeletal clone option which uses the blank template as the source to avoid copying gigabytes of data that will be deleted.  https://github.com/ManageIQ/ovirt/blob/master/lib/ovirt/template.rb#L109

Comment (matobet) [inline lib/ovirt/template.rb]: @bdunne Ok, but I don't see how this would be negatively affected by this change, since blank template should have the default boot order of `hd` so your skeletal clone should IMO still work.

Comment (bdunne) [inline lib/ovirt/template.rb]: @matobet During provisioning if you select a template to use as the base for PXE or ISO provisioning, I would expect all attributes (boot order included) to be inherited from this template.  Behind the scenes, since we use the blank template as the source of the clone, those values wouldn't be inherited by default and we need to set them.  The defaults would be inherited from the blank template instead of the selected template (which might have a different boot order).

**Comment by Fryguy** [comment]
@matobet Please fix the rubocop issues.

Also, not sure how this will affect EMS Refresh in ManageIQ, but I assume you are on top of that.

@bdunne Can you please review and merge if you're good?

File: .gitignore [hunk 2]
```diff
@@ -1,7 +1,9 @@
 *.gem
+*.iml
 *.rbc
 .bundle
 .config
+.idea
```
Comment (gmcculloug) [inline .gitignore]: @matobet Was this include by mistake?  I am not against excluding these file types but think it should be a separate PR.

Comment (matobet) [inline .gitignore]: @gmcculloug Ok, I'll split this from the patch.

**Comment by matobet** [comment]
@Fryguy @bdunne but please don't merge just yet. I still need to verify this new version (and whether it actually covers the discussed PXE/ISO provisioning)

**Comment by bdunne** [comment]
@matobet Thanks for investigation this.  Can you create a test for the PXE / ISO case where the selected template has a non-default boot order and make sure that the resulting VM has the same boot order?

**Comment by gmcculloug** [comment]
@matobet Please correct the syntax errors that are causing tests to fail.

@bdunne Assuming those errors are the only thing stopping the tests from passing please review the latest changes.

**Comment by bdunne** [comment]
@gmcculloug Yes, aside from the Rubocop and Travis errors, it looks good.
@matobet Thanks for adding those tests. :icecream:

**Comment by matobet** [comment]
@bdunne no problem, I certainly learned a lot when making those tests :+1:. And sorry for the syntax errors. For some reason I assumed that the baseline ruby version is the same as for ManageIQ (2.2+) :smiley:

**Comment by gmcculloug** [comment]
@bdunne Since this is merged we should remove the "[WIP]" from the title.  😃  (I do not have permission to do it.)
